### PR TITLE
[docs] Documents openwisp-utils-qa-checks #47

### LINKED ISSUE #47: [docs] Document openwisp-utils-qa-check
Add the description of `checkendline` and `openwisp-utils-qa-check` to the README, in a similar way to how the other quality assurance scripts are documented.

### PR DESCRIPTION
**Pull request by TPath123**
Adds openwisp-utils-qa-checks to the README file and describes its purpose. It also includes `checkendline`

Fixes #47

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -305,6 +305,40 @@ Usage example::
...
+~~~~~~~~~~~~~~~~~
+
+Ensures that a blank line is kept at the end of each file.
+
+We want to keep diffs as clean as possible, and this allows us to append further changes to
+the end of a file without a messy diff.
+
+This is also important for some tools, so that reading from an input stream knows when to terminate.
+
+Usage example::
```
Comment (atb00ker) [inline README.rst]: If it can't be used like this, then why do we have a "Usage example"?

**Comment by TPath123** [comment]
@atb00ker Ah I see, it must’ve been added after I made my commit (was away for a few days so couldn’t push immediately)
For your final comment, I did state that “all of the above options can be used inside qa-checks”, as they are all documented previously - should I make it more explicit then?

File: README.rst (hunk 1 above)
Comment (TPath123) [reply README.rst]: Apologies, I think I left it there because at first I was looking for a command line command like the rest, I’ll clean this up

**Comment by atb00ker** [comment]
> For your final comment, I did state that “all of the above options can be used inside qa-checks”, as they are all documented previously - should I make it more explicit then?

Some options are not documented, please read the code & testcases for all the tests, it should help you find out, when you think you've found all of them, mark for review and I'll double check to let you know! :smile:

File: README.rst [hunk 2]
```diff
@@ -270,6 +270,28 @@ automated builds of different OpenWISP modules.
 ``openwisp-utils-qa-checks``
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Runs all quality assurance checks in a single file
```
Comment (atb00ker) [inline README.rst]: In single file, which file?
Change this with:
```
``openwisp-utils-qa-checks``
~~~~~~~~~~~~~~~~~~~~~~
Shell script to run the following quality assurance checks:

* ``checkmigrations`` <link-to-section>
* ``checkcommit`` <link-to-section>
* ``checkendline`` <link-to-section>
* ``flake8`` - Python code linter
* ``isort`` - Sorts python imports alphabetically, and seperated into sections
```

File: README.rst [hunk 3]
```diff
@@ -270,6 +270,28 @@ automated builds of different OpenWISP modules.
...
+
+The following quality assurance checks are ran in openwisp-utils-qa-checks:
+
+* ``checkmigrations`` (Read more below)
+* ``checkcommit`` (Read more below)
```
Comment (atb00ker) [inline README.rst]: This [line](https://github.com/openwisp/openwisp-utils/blob/master/tests/test_project/tests/test_qa.py#L105) "Simulation of a special unplanned case\n\n#noqa" shows an undocumented change. Please document `#noqa` case.

File: README.rst [hunk 4]
```diff
@@ -323,6 +345,18 @@ Usage example::
     Simulation of a special unplanned case
     #noqa
 
+``checkendline``
+~~~~~~~~~~~~~~~~~
+
+Ensures that a blank line is kept at the end of each file.
+
+We want to keep diffs as clean as possible, and this allows us to append further changes to
```
Comment (atb00ker) [inline README.rst]: What's diffs? Please remove this line! :smile:

File: README.rst [hunk 5]
```diff
@@ -323,6 +345,18 @@ Usage example::
...
+the end of a file without a messy diff.
+
+This is also important for some tools, so that reading from an input stream knows when to terminate.
+
+``checkendline`` is used as a part of ``openwisp-utils-qa-checks``
```
Comment (atb00ker) [inline README.rst]: Not needed.

File: README.rst (hunk 3 above)
Comment (TPath123) [reply README.rst]: @atb00ker Could you explain what you mean here? I believe the `#noqa` case is documented above?

Comment (atb00ker) [reply README.rst]: Can you please send the link?

**Pull request merged**